[1차]-다트-게임.js

```diff
@@ -0,0 +1,19 @@
+function solution(dartResult) {
+    let newResult = dartResult
+        .replace(/\*/g, " *2 ")
+        .replace(/S/g, "**1 ")
+        .replace(/D/g, "**2 ")
+        .replace(/T/g, "**3 ")
+        .replace(/\#/g, "(-1) ")
+        .replace(/ \(/g, "(")
+        .replace(/ {1,}\*/g, "*")
+        .replace(/ {1,}$/g, "");
+    let newArray = newResult.split(" ");
+    newArray = newArray.forEach((item, index) => {
+      if (item.match(/\*{1,}/g).length > 1 && index > 0) {
+          newArray[index - 1] = newArray[index - 1] + '*2'
+        }
+    });
+    const result = newArray.join(' ').replace(/ /g, '+').replace(/\(/g, '*(');
+    return (eval(result));
+}
```

Comment:
```suggestion
function solution(dartResult) {
    const newArray = dartResult
        .replace(/\*/g, " *2 ")
        .replace(/S/g, "**1 ")
        .replace(/D/g, "**2 ")
        .replace(/T/g, "**3 ")
        .replace(/\#/g, "(-1) ")
        .replace(/ \(/g, "(")
        .replace(/ {1,}\*/g, "*")
        .replace(/ {1,}$/g, "")
        .split(" ");
    
    newArray.forEach((item, index) => {
      if (item.match(/\*{1,}/g).length > 1 && index > 0) {
          newArray[index - 1] = newArray[index - 1] + '*2'
        }
    });
    
    const result = newArray.join(' ').replace(/ /g, '+').replace(/\(/g, '*(');
    
    return (eval(result));
}
```
제출한 코드는 로직은 맞는거같은데 테스트가 통과가 안돼요. forEach 메서드는 반환값이 없고 해당 배열의 요소들을 바꾸기만 하기 떄문에 통과가 되게 하려면 저런식으로 코드가 짜여져야합니다.